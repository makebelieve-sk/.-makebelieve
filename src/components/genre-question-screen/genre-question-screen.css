.game {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0;
    padding: 0;
    width: 100%;
    height: 100%; 
}
.game_screen {
    position: relative;
    z-index: 1;
    width: 440px;
    text-align: center; 
}
.game_title {
    margin: 0;
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    font-size: 33px;
    color: #230d1a;
    margin-bottom: 20px;
    margin-top: 20px; 
} 
.game_track {
    width: 100%;
    display: flex;
    margin-bottom: 10px; 
}
.game_answer {
    margin-left: 15px;
    width: 100px;
    height: 80px; 
    display: flex;  
}
.game_check {
    margin-left: 20px;
    width: 57px;
    height: 80px;
    background: url('./img/icon-note-inactive.png');
    cursor: pointer;
}
.game_input {
    opacity: 0;
}
input[type="checkbox"]:checked + .game_check {
    background-image: url("./img/icon-note-active.png"); 
}
.game_submit {
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    margin-top: 30px;
    padding: 5px 20px;
    font-size: 24px; 
    cursor: pointer;
}