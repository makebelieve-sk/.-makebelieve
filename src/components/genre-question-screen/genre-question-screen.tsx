import React from 'react';

import { isArtistAnswerType, userAnswerType, questionType, isGenreAnswerType } from '../../types';

import './genre-question-screen.css';

type GenreQuestionScreenType = {
    question: questionType,
    onAnswer: (userAnswerArray: any) => any,
    renderPlayer: (answer: isGenreAnswerType, i: number) => any,
    getUserAnswer: (answer: isGenreAnswerType, i: number) => any,
    userAnswerArray: isArtistAnswerType | userAnswerType
};

export const GenreQuestionScreen: React.FC<GenreQuestionScreenType> = ({ question, onAnswer, renderPlayer, getUserAnswer, userAnswerArray }) => {
    const { genre, answers } = question;

    return (
        <section className="game">
            <section className="game_screen">
                <h2 className="game_title">Выберите { genre } треки:</h2>
                <form className="game_tracks" onSubmit={(e) => {
                    e.preventDefault();
                    onAnswer(userAnswerArray);
                }}>
                    {answers.map((answer, i) => {
                        return (
                            <div className="game_track" key={`answer-${answer.id}`}>
                                {renderPlayer(answer, i)}
                                {getUserAnswer(answer, i)}
                            </div>
                        );
                    })}
                    <button className="game_submit button" type="submit">Ответить</button>
                </form>
            </section>
        </section>
    );
};