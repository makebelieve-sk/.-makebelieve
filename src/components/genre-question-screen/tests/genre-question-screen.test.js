import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { GenreQuestionScreen } from '../genre-question-screen';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2, prop3, prop4, prop5) => shallow(
    <GenreQuestionScreen question={prop1} onAnswer={prop2} renderPlayer={prop3} getUserAnswer={prop4} userAnswerArray={prop5} />
);

const mock = {
    question: {
        song: {
            artist: `John Snow`,
            src: `https://s3.amazonaws.com/freecodecamp/drums/Heater-1.mp3`
        },
        answers: [
            {
                id: 1,
                picture: `https://placehold.it/134x134`,
                artist: 'John Snow'
            },
            {
                id: 2,
                picture: `https://placehold.it/134x134`,
                artist: 'Jack Daniels'
            },
            {
                id: 3,
                picture: `https://placehold.it/134x134`,
                artist: 'Jim Beam'
            }
        ]
    },
    onAnswer: jest.fn(),
    renderPlayer: jest.fn(),
    getUserAnswer: jest.fn(),
    userAnswerArray: [false, false, false, false]
};

describe(`GenreQuestionScreen component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.question, mock.onAnswer, mock.renderPlayer, mock.getUserAnswer, mock.userAnswerArray);
    });

    it(`should contain section wrapper`, () => {
        const wrapper = component.find('.game');
        expect(wrapper.length).toBe(1);
    });

    it(`GenreQuestionScreen correctly renders after relaunch`, () => {
        const tree = renderer  
            .create(<GenreQuestionScreen 
                question={mock.question}
                onAnswer={mock.onAnswer}
                renderPlayer={mock.renderPlayer}
                getUserAnswer={mock.getUserAnswer}
                userAnswerArray={mock.userAnswerArray}
            />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});