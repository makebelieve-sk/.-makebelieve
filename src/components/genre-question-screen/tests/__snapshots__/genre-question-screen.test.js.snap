// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreen component GenreQuestionScreen correctly renders after relaunch 1`] = `
<section
  className="game"
>
  <section
    className="game_screen"
  >
    <h2
      className="game_title"
    >
      Выберите 
       треки:
    </h2>
    <form
      className="game_tracks"
      onSubmit={[Function]}
    >
      <div
        className="game_track"
      />
      <div
        className="game_track"
      />
      <div
        className="game_track"
      />
      <button
        className="game_submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
